src/main/java/com/google/sps/data/User.java

```diff
@@ -14,8 +14,30 @@
 
 package com.google.sps.data;
 
-public class Cool {
-  public static String getCool() {
-    return "Cool";
+public class User {
+  private final Long id;
+  private final String email;
+  private final String nickname;
+
+  public User(Long id, String email, String nickname) {
+    if (id == null) {
+      throw new IllegalArgumentException("id cannot be null");
+    }
+
+    if (email == null) {
+      throw new IllegalArgumentException("email cannot be null");
+    }
+
+    if (nickname == null) {
+      throw new IllegalArgumentException("nickname cannot be null");
+    }
+
+    this.id = id;
+    this.email = email;
+    this.nickname = nickname;
```

Comment:
Optional: You can simplify this by using https://docs.oracle.com/javase/8/docs/api/java/util/Objects.html#requireNonNull-T-java.lang.String-